sdk/appconfiguration/app-configuration/test/public/utils/testHelpers.ts

```diff
@@ -163,9 +162,6 @@ export async function assertThrowsRestError(
     await testFunction();
     assert.fail(`${message}: No error thrown`);
   } catch (err) {
-    if (!(err instanceof RestError)) {
```

Comment:
Did the pipeline fail because of this line should assert `Error` instead? The other two look fine to me if we expect them to be `Error`.  Since the code is already merged in main, we could leave them there if they are correct, which would save us from doing it again later when moving to TypeScript v4.4

Reply:
min-max pipelines were failing, which is why this PR was needed in the first place.

Reply:
I suspect that we were getting some unexpected errors in the min-max testing.  It should have passed this check if it's expected. Even if we removed this check, it would fail below.  I was thinking that we might want to do `instance of Error` check here, and let it fail later.

Reply:
The main problem with the pipelines was related to this check, err was not an instance of `RestError`. 
I can try checking for `instance of Error` and see if we don't get the "Error is not recognized" on the pipeline

Reply:
I suspect that it is caused by two different versions of core-http in min/max testing. Pushed an update. This hopefully should fix the build.